osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -540,7 +540,7 @@ public void SelectPreviousRandom()
             {
                 var beatmap = randomSelectedBeatmaps.Pop();
 
-                if (!beatmap.Filtered.Value)
+                if (!beatmap.Filtered.Value && beatmapSets.Any(beatset => beatset.Beatmaps.Contains(beatmap)))
```

Comment:
Should probably not be checking all beatmaps like that and be doing

```suggestion
                if (!beatmap.Filtered.Value && beatmap.BeatmapSet?.PendingDelete != false)
```

instead.

Reply:
The only way to access the `PendingDelete` flag would be by: `beatmap.BeatmapInfo.BeatmapSet.PendingDelete`. And apparently, that doesn't look to get updated when the beatmap is actually deleted.

I've went with a different solution that's still magnitudes more efficient than iterating through all beatmaps in the carousel.